ocean_lib/data_provider/data_service_provider.py

```diff
@@ -68,6 +68,34 @@ def encrypt_files_dict(files_dict, encrypt_endpoint, asset_id, publisher_address
 
             return response.json()['encryptedDocument']
 
+    @staticmethod
+    def check_url(url, check_url_endpoint):
+        payload = json.dumps({'url': url})
+
+        response = DataServiceProvider._http_client.post(
+            check_url_endpoint, data=payload,
+            headers={'content-type': 'application/json'}
+        )
+
+        if (
+            not response or
+            not hasattr(response, 'status_code') or
+            response.status_code != 200
+        ):
+            msg = (f'Could not determine Content-Length and Content-Type '
+                   f'{check_url_endpoint}, status {response.status_code}')
+            logger.error(msg)
+            raise OceanEncryptAssetUrlsError(msg)
+
+        result = response.json()['result']
```

Comment:
why is it {'result': value}, can it just be `value` which is a dict?

Reply:
that's how the endpoints seemed to be structured on provider: either a "result" with the actual result inside, or an "error" with the error description. I inspired myself from the previous endpoints and it seemed to make sense. Do you want me to change the provider endpoint to return the result directly?

Reply:
if it's consistent with the response from other provider endpoints then  let's keep it like this.